catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/DeleteOperations.java

```diff
@@ -311,6 +316,44 @@ public DeleteResponse delete(DeleteRequest deleteRequest)
     //
     // Private helper methods
     //
+    private DeleteRequest rewriteQuery(DeleteRequest deleteRequest)
+            throws UnsupportedQueryException {
+        if (Metacard.ID.equals(deleteRequest.getAttributeName())) {
```

Comment:
Might want to store the value of `deleteRequest.getAttributeName()` in a local variable since you use it in a few places.